src/picongpu/include/particles/bremsstrahlung/ScaledSpectrum.hpp

```diff
@@ -0,0 +1,312 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "cuSTL/container/HostBuffer.hpp"
+#include "cuSTL/cursor/Cursor.hpp"
+#include "cuSTL/cursor/navigator/PlusNavigator.hpp"
+#include "cuSTL/cursor/tools/LinearInterp.hpp"
+#include "cuSTL/cursor/BufferCursor.hpp"
+#include "algorithms/math.hpp"
+#include <boost/array.hpp>
+#include <boost/numeric/odeint/integrate/integrate.hpp>
+#include <boost/shared_ptr.hpp>
+#include <limits>
+
+namespace picongpu
+{
+namespace particles
+{
+namespace bremsstrahlung
+{
+
+namespace detail
+{
+
+/** Functor for the scaled differential cross section (dcs) which
+ * equals to the electron energy loss times the cross section per unit energy.
+ */
+struct LookupTableFunctor
+{
+    typedef typename ::PMacc::result_of::Functor<
+        ::PMacc::cursor::tools::LinearInterp<float_X>,
+        ::PMacc::cursor::BufferCursor<float_X, DIM2> >::type LinInterpCursor;
+
+    typedef float_X type;
+
+    LinInterpCursor linInterpCursor;
+    float_X lnEMin;
+    float_X lnEMax;
+
+    /** constructor
+     *
+     * @param linInterpCursor
+     */
+    HDINLINE LookupTableFunctor(LinInterpCursor linInterpCursor)
+        : linInterpCursor(linInterpCursor)
+    {
+        const float_64 EMin_SI = electron::MIN_ENERGY_MeV * 1.0e3 * UNITCONV_keV_to_Joule;
+        const float_64 EMax_SI = electron::MAX_ENERGY_MeV * 1.0e3 * UNITCONV_keV_to_Joule;
+
+        this->lnEMin = algorithms::math::log(EMin_SI / UNIT_ENERGY);
+        this->lnEMax = algorithms::math::log(EMax_SI / UNIT_ENERGY);
+    }
+
+    /** scaled differential cross section
+     *
+     * @param Ekin kinectic energy of the incident electron
+     * @param kappa energy loss normalized to Ekin
+     */
+    HDINLINE float_X operator()(const float_X Ekin, const float_X kappa) const
+    {
+        const float_X lnE = algorithms::math::log(Ekin);
+
+        const float_X binE = (lnE - this->lnEMin) / (this->lnEMax - this->lnEMin) * static_cast<float_X>(electron::NUM_SAMPLES_EKIN - 1);
+        // in the low-energy limit Bremsstrahlung is not taken into account
+        if(binE < float_X(0.0))
+            return float_X(0.0);
+        const float_X binKappa = kappa * static_cast<float_X>(electron::NUM_SAMPLES_KAPPA - 1);
+
+        if (picLog::log_level & picLog::CRITICAL::lvl)
+        {
+            if(binE > float_X(electron::NUM_SAMPLES_EKIN - 1))
+            {
+                const float_64 Ekin_SI = Ekin * UNIT_ENERGY;
+                printf("[Bremsstrahlung] error lookup table: Ekin=%g MeV is out of range.\n",
+                       float_X(Ekin_SI * UNITCONV_Joule_to_keV * 1.0e-3));
+            }
+            if(binKappa < float_X(0.0) || binKappa > float_X(electron::NUM_SAMPLES_KAPPA - 1))
```

Comment:
here a check for `kappa < 0.0 || kappa > 1.0` would be more intuitive **and** you could avoid a cast.